src/oisst/State/State.cc

```diff
@@ -5,120 +5,366 @@
  * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
+#include <numeric>
+#include <cmath>
+#include <string>
+#include <vector>
+#include <iostream>
+#include <iterator>
+
+// #include <netcdf>
+#include "netcdf"
+
 #include "oisst/Geometry/Geometry.h"
 #include "oisst/State/State.h"
 
 #include "eckit/config/Configuration.h"
 
+#include "atlas/field.h"
+#include "atlas/array.h"
+#include "atlas/option.h"
+
 #include "oops/base/Variables.h"
 #include "oops/util/abor1_cpp.h"
+#include "oops/util/Logger.h"
 
 #include "ufo/GeoVaLs.h"
 #include "ufo/Locations.h"
 
+// using namespace netCDF;
+using atlas::array::make_view;
+using atlas::option::name;
+
+
 namespace oisst {
 
 // ----------------------------------------------------------------------------
 
   State::State(const Geometry & geom, const eckit::Configuration & conf)
     : geom_(new Geometry(geom)), time_(conf.getString("date")),
       vars_(conf, "state variables") {
-    util::abor1_cpp("State::State() needs to be implemented.",
-                     __FILE__, __LINE__);
+
+    if (vars_.size() != 1) {
+      util::abor1_cpp("State::State() needs to be implemented.",
```

Comment:
I would recommend replacing the message here, and in the other `vars_.size() != 1` sections with a more instructive message.... jus in case

Reply:
Done.